apps/github-api-integration/behaviours/fetch-repos.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Model = require('../models/github');
+const _ = require('lodash');
+
+// Behaviour wrapped in a function here
+// allows us to pass in a config
+module.exports = config => {
+
+  return superclass => class extends superclass {
+
+    // `getValues` is part of the GET pipeline. For details
+    // see https://github.com/UKHomeOfficeForms/hof-behaviour-hooks.
+    getValues(req, res, next) {
+
+      // Get the user entered input previously
+      // stored on the session model
+      const name = req.sessionModel.get('name');
+
+      // Define an instance of the Model
+      // with attributes and config (uri)
+      const model = new Model({name}, config);
+
+      // Call `fetch` to get the resource
+      // from the url we configured
+      model.fetch().then(result => {
+
+          // Create a subset of the result
+          const repos = _.map(result, repo => ({
+            name: repo.name,
+            desc: repo.description,
+            url: repo.html_url
+          }));
+
+          // Set the user name to the locals
+          res.locals.name = name;
```

Comment:
Rather than getting this field explicitly from the session model, I'd be tempted to call up to `super.getValues` and then augment the callback value from that (which will include `req.sessionModel.toJSON()`) with the repos.

That way the name can be accessed in the template at `{{values.name}}`.

We probably want to call through to the `super` anyway to avoid providing unexpected values to any upstream behaviours.